src/opentaskpy/addons/gcp/remotehandlers/bucket.py

```diff
@@ -0,0 +1,304 @@
+                    params={"alt": "media"},  # Remove to only grab obj metadata
+                )
+                if response.status_code == 401:
+                    self.logger.error(f"Unauthorized to GET file: {file}")
+                    result = 1
+                elif response.status_code == 403:
+                    self.logger.error(f"Failed to GET file: {file}")
+                    self.logger.error(f"Forbidden Status Code: {response.status_code}")
+                    result = 1
+                elif not response.ok:
+                    self.logger.error(f"Failed to GET file: {file}")
+                    self.logger.error(f"Got return code: {response.status_code}")
+                    self.logger.error(response)
+                    result = 1
+                else:
+                    with open(
+                        f"{local_staging_directory}/{file.split('%2F')[-1]}", "wb"
+                    ) as f:
+                        f.write(response.content)
+                    self.logger.info(
+                        f"Successfully downloaded {file} to local Staging directory"
+                    )
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            self.logger.error(f"Failed to download file: {file}")
+            self.logger.exception(e)
+            result = 1
+
+        return result
+
+    def transfer_files(
+        self,
+        files: list[str],
+        remote_spec: dict,
+        dest_remote_handler: RemoteTransferHandler,
+    ) -> int:
+        """Not implemented for this transfer type."""
+        raise NotImplementedError
+
+    def create_flag_files(self) -> int:
+        """Not implemented for this transfer type."""
+        raise NotImplementedError
+
+    def list_files(
+        self, directory: str | None = None, file_pattern: str | None = None
+    ) -> list:
+        self.logger.info("Listing Files method.")
+        try:
+            file_pattern = self.spec["fileExpression"]
+            if "directory" in self.spec and self.spec["directory"] != "":
+                file_pattern = f"{self.spec['directory']}/{file_pattern}"
+
+            response = requests.get(
+                f"https://storage.googleapis.com/storage/v1/b/{self.spec['bucket']}/o",
+                headers={"Authorization": f"Bearer {self.credentials}"},
+                timeout=1800,
+                params={"matchGlob": file_pattern},
+            )
+            if response.status_code == 200:
+                data = response.json()
+                if "items" in data:
+                    items = data["items"]
+                    names = [item["name"] for item in items if "name" in item]
+                    return names
+                else:
+                    self.logger.info(
+                        f"No items which matches {file_pattern} found in directory."
+                    )
+                    return []
+            else:
+                self.logger.error(
```

Comment:
If you're here, should it not be generating an exception to tell whatever called it that there was a problem? Otherwise it might get stuck in a loop of errors trying to list files over and over but never working?